devtools/common/src/main/java/io/quarkus/templates/TemplateRegistry.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ * <p>
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package io.quarkus.templates;
+
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * @author <a href="claprun@redhat.com">Christophe Laprun</a>
+ */
+public class TemplateRegistry {
+
+    private static final Map<String, QuarkusTemplate> templates = new ConcurrentHashMap<>(7);
+    private static final TemplateRegistry INSTANCE = new TemplateRegistry();
+
+    private TemplateRegistry() {
+        loadTemplates();
+    }
+
+    public static TemplateRegistry getInstance() {
+        return INSTANCE;
+    }
+
+    public static QuarkusTemplate createTemplateWith(String name) throws IllegalArgumentException {
+        final QuarkusTemplate template = templates.get(name);
+        if (template == null) {
+            throw new IllegalArgumentException("Unknown template: " + name);
+        }
+
+        return template;
+    }
+
+    private static void register(QuarkusTemplate template) {
+        if (template != null) {
```

Comment:
Shouldn't it throw a runtime exception if the template is null?

Reply:
Probably, yes.